app/javascript/src/NewService/components/NewServiceForm.jsx

```diff
@@ -0,0 +1,120 @@
+// @flow
+
+import React, {useState, useEffect} from 'react'
+import {fetch as fetchPolyfill} from 'whatwg-fetch'
+import 'core-js/es6/promise'
+import {ServiceSourceForm, FormTemplate} from 'NewService'
+import type {SourceFormProps, FormTemplateProps} from 'NewService/types'
+
+import {createReactWrapper} from 'utilities/createReactWrapper'
+
+const BASE_PATH = '/p/admin/service_discovery'
+const PROJECTS_PATH = `${BASE_PATH}/projects.json`
+
+type Props = {
+  isServiceDiscoveryAccesible: boolean,
+  isServiceDiscoveryUsable: boolean,
+  serviceDiscoveryAuthenticateUrl: string,
+  providerAdminServiceDiscoveryServicesPath: string,
+  adminServicesPath: string
+}
+
+type Response = {
+  ok: boolean,
+  statusText: string,
+  json: () => Promise<void>
+}
+const fetchData = async (url: string, setError) => {
+  setError({hasError: false, errorMsg: ''})
+  const namespace = url === PROJECTS_PATH
+    ? 'projects'
+    : 'services'
+
+  return fetchPolyfill(url)
+    .then((response: Response) => {
+      if (!response.ok) {
+        setError({hasError: true, errorMsg: response.statusText})
+      }
+      return response.json()
+    })
+    .then(data => data[namespace].reduce((acc, current) => acc.concat(current.metadata.name), []))
+    .catch(error => {
+      console.error(error)
+    })
+}
+
+const callEfect = (servicesRequested, selectedNamespace, startServicesFetch) => {
+  if (servicesRequested && selectedNamespace) {
+    const SERVICES_PATH = `${BASE_PATH}/namespaces/${selectedNamespace}/services.json`
```

Comment:
generally one uses CAPITAL for constant that are meant for a main/general setting/value that won't change and will always be the same value. Here you could just call it `servicePath` or call `startServicesFetch` with the strng template